src/presets/firebase.ts

```diff
@@ -8,15 +8,28 @@ import { writeFile } from "../utils";
 import { defineNitroPreset } from "../preset";
 import type { Nitro } from "../types";
 
+// default to one for retro compatibility but should be set to 2
+const firebaseFunctionsGen = process.env.FIREBASE_FUNCTIONS_GEN || "1";
+
+if (firebaseFunctionsGen !== "1" && firebaseFunctionsGen !== "2") {
+  throw new Error(
+    `FIREBASE_FUNCTIONS_GEN must be 1 or 2, got "${firebaseFunctionsGen}"`
+  );
+}
+
 export const firebase = defineNitroPreset({
-  entry: "#internal/nitro/entries/firebase",
+  entry: `#internal/nitro/entries/firebase-gen-${firebaseFunctionsGen}`,
```

Comment:
Tip: We can read it from input config too. Just make preset a factory function!

Reply:
aah, much better, thanks!

Reply:
How do I read the nitro options from the factory function? It seems to not have any argument

Reply:
Oh right sorry i forgot presets are layers. We have an alternative. `entry` also supports special templates ([src](https://github.com/unjs/nitro/blob/fddf57e50ed4bd4863a23d80c86645bcea79b884/src/utils/index.ts#L72)). Passing `entry: #internal/nitro/entries/firebase-gen-{{ firebase.gen }}` should do the trick (please note that we need to define default gen in `options.ts` defaults if there is no value, template is not replaced). I will improve it afterwards

Reply:
Can we set a default value in the preset directly ?

Reply:
Please try. Yes we might if c12 merges correctly.